### DISCUSSION TITLE
Using LogQL JSON outputs from Loki in Dashboards

### DISCUSSION THREAD

**Discussion by esbeneickhardt** [grafana/grafana#50318] [Dashboards, Panels & Visualizations]
After getting familiar with the LogQL query language, I have cleaned up my logs such that I have a bit of JSON pr line:

```
{"Name": "App1", "VulnerabilityID": "89", "Vulnerability": "CRITICAL"}
{"Name": "App1", "VulnerabilityID": "134", "Vulnerability": "LOW"}
{"Name": "App2", "VulnerabilityID": "25", "Vulnerability": "LOW"}
```

Now I would like to make use of Grafana's cool visuals and tables, but Grafana does not seem to be able to extract the values correctly from the JSON. If JSON is pulled from an API or some other data source everything works smoothly.

Do you have any suggestions for how to use Grafana visualizations with Loki JSON-logs, or do I really have to place the logs in a database first?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
